### PR TITLE
Use --format='value(name)' with gcloud instead of grep/awk/cut

### PR DESCRIPTION
**Pull request by ixdy**
Fixing our fragile parsing of `gcloud` is getting old (#24746, #25159, maybe others?).

Instead, let's just get the proper output out of `gcloud` in the first place.

### CODE REVIEW

**Comment by ixdy** [comment]
cc @jlowdermilk @roberthbailey I think the GKE stuff is correct (it seems to work), but I'd appreciate a look-over.

File: cluster/gke/util.sh [hunk 1]
```diff
@@ -190,7 +190,7 @@ function test-setup() {
   detect-nodes >&2
 
   # At this point, CLUSTER_NAME should have been used, so its value is final.
-  NODE_TAG=$($GCLOUD compute instances describe ${NODE_NAMES[0]} --project="${PROJECT}" --zone="${ZONE}" | grep -o "gke-${CLUSTER_NAME}-.\{8\}-node" | head -1)
+  NODE_TAG=$($GCLOUD compute instances describe ${NODE_NAMES[0]} --project="${PROJECT}" --zone="${ZONE}" --format='value(tags.items)' | grep -o "gke-${CLUSTER_NAME}-.\{8\}-node")
```
Comment (ixdy) [inline cluster/gke/util.sh]: I don't know if we still need the `grep` here. Right now I think the node tag is the only thing in here, but I'm not sure that will always be the case.

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -895,28 +895,13 @@ function kube-down {
   # change during a cluster upgrade.)
   local template=$(get-template "${PROJECT}")
 
-  # The gcloud APIs don't return machine parseable error codes/retry information. Therefore the best we can
-  # do is parse the output and special case particular responses we are interested in.
   for group in ${INSTANCE_GROUPS[@]:-}; do
     if gcloud compute instance-groups managed describe "${group}" --project "${PROJECT}" --zone "${ZONE}" &>/dev/null; then
-      deleteCmdOutput=$(gcloud compute instance-groups managed delete --zone "${ZONE}" \
+      gcloud compute instance-groups managed delete \
```
Comment (jlowdermilk) [inline cluster/gce/util.sh]: Was this command changed to wait for the operation to complete?

File: cluster/gke/util.sh (hunk 1 above)
Comment (jlowdermilk) [inline cluster/gke/util.sh]: Let's keep it. We might add tags in future, plus users can put tags on node vms.

File: cluster/gce/util.sh (hunk 2 above)
Comment (ixdy) [inline cluster/gce/util.sh]: it appears to.

**Comment by jlowdermilk** [comment]
lgtm, but please squash commits.

**Comment by roberthbailey** [comment]
@ixdy can you squash your commits?

**Comment by ixdy** [comment]
rebased and squashed.

**Comment by roberthbailey** [comment]
Re-applying LGTM after squash.

**Comment by ixdy** [comment]
FYI @matchstick @lavalamp we may want to cherry-pick this to 1.2.

**Comment by lavalamp** [comment]
@ixdy is this one essential or is #24746 sufficient?

**Comment by ixdy** [comment]
We can try that one to start. This PR was more proactive instead of reactive, but it's possible that gcloud has changed the formatting of other things since this PR was merged.

**Pull request merged**